localization/pose_estimator_arbiter/src/pose_estimator_arbiter/pose_estimator_arbiter_core.cpp

```diff
@@ -0,0 +1,202 @@
+
+    // Subscriber for stoppers
+    sub_artag_input_ = create_subscription<Image>(
+      "~/input/artag/image", sensor_qos, shared_data_->artag_input_image.create_callback());
+    sub_yabloc_input_ = create_subscription<Image>(
+      "~/input/yabloc/image", sensor_qos, shared_data_->yabloc_input_image.create_callback());
+    sub_ndt_input_ = create_subscription<PointCloud2>(
+      "~/input/ndt/pointcloud", sensor_qos, shared_data_->ndt_input_points.create_callback());
+    sub_eagleye_output_ = create_subscription<PoseCovStamped>(
+      "~/input/eagleye/pose_with_covariance", 5,
+      shared_data_->eagleye_output_pose_cov.create_callback());
+
+    // Subscriber for switch rule
+    sub_localization_pose_cov_ = create_subscription<PoseCovStamped>(
+      "~/input/pose_with_covariance", 5, shared_data_->localization_pose_cov.create_callback());
+    sub_point_cloud_map_ = create_subscription<PointCloud2>(
+      "~/input/pointcloud_map", latch_qos, shared_data_->point_cloud_map.create_callback());
+    sub_vector_map_ = create_subscription<HADMapBin>(
+      "~/input/vector_map", latch_qos, shared_data_->vector_map.create_callback());
+    sub_initialization_state_ = create_subscription<InitializationState>(
+      "~/input/initialization_state", latch_qos,
+      shared_data_->initialization_state.create_callback());
+  }
+
+  // Load switching rule
+  load_switch_rule();
+
+  // Timer callback
+  auto on_timer = std::bind(&PoseEstimatorArbiter::on_timer, this);
+  timer_ =
+    rclcpp::create_timer(this, this->get_clock(), rclcpp::Rate(1).period(), std::move(on_timer));
+
+  // Enable all pose estimators at the first
+  toggle_all(true);
+}
+
+void PoseEstimatorArbiter::load_switch_rule()
+{
+  // NOTE: In the future, some rule will be laid below
+  RCLCPP_INFO_STREAM(get_logger(), "load default switching rule");
+  switch_rule_ =
+    std::make_shared<switch_rule::MapBasedRule>(*this, running_estimator_list_, shared_data_);
+}
+
+void PoseEstimatorArbiter::toggle_each(
+  const std::unordered_map<PoseEstimatorType, bool> & toggle_list)
+{
+  for (auto s : stoppers_) {
+    if (toggle_list.at(s.first)) {
+      s.second->enable();
+    } else {
+      s.second->disable();
+    }
+  }
+}
+
+void PoseEstimatorArbiter::toggle_all(bool enabled)
+{
+  std::unordered_map<PoseEstimatorType, bool> toggle_list;
+  for (auto s : stoppers_) {
+    toggle_list.emplace(s.first, enabled);
+  }
+  toggle_each(toggle_list);
+}
+
+void PoseEstimatorArbiter::publish_diagnostics() const
+{
+  diagnostic_msgs::msg::DiagnosticStatus diag_status;
+
+  // Temporary implementation
+  {
+    diag_status.name = "localization: " + std::string(this->get_name());
+    diag_status.hardware_id = this->get_name();
+
+    diag_status.level = diagnostic_msgs::msg::DiagnosticStatus::OK;
+    diag_status.message = "OK";
+
+    diagnostic_msgs::msg::KeyValue key_value_msg;
+    key_value_msg.key = "state";
+    key_value_msg.value = "Diagnostics is not implemented.";
```

Comment:
What does "Diagnostics is not implemented." mean here? (It IS implemented after all).
Should I take this as "Further details to be implemented later."?

Reply:
I meant that the details were unimplemented. I fixed the message in https://github.com/autowarefoundation/autoware.universe/pull/5846/commits/80b6c20eef552e1a3b5176c1bebc5c43da95b827

Reply:
Thank you for your prompt response!
Seems good to me. :+1: